src/raiden_libs/blockchain.py

```
@@ -35,15 +36,31 @@
 log = structlog.get_logger(__name__)
 
 
+def get_web3_provider_info(web3: Web3) -> str:
+    """ Returns information about the provider
+
+    Currently works only with `HTTPProvider`. Needs to be adapted when new procviders
+    are added.
+    """
+    provider = web3.provider
+    if isinstance(provider, HTTPProvider):
+        endpoint = provider.endpoint_uri
+        if endpoint is not None:
+            return str(endpoint)
+
+    return ""
```

Should we raise an error here, instead?

---

Good idea